mitiq/interface/mitiq_qiskit/conversions.py

```diff
@@ -255,6 +255,9 @@ def from_qiskit(circuit: qiskit.QuantumCircuit) -> cirq.Circuit:
         # Try to decompose circuit before running
         # This is necessary for converting qiskit circuits with
         # custom packaged gates, e.g., QFT gates
+        circuit = circuit.decompose(
+            gates_to_decompose=["u1", "u2", "u3", "cx", "rx", "ry"]
```

Comment:
First, thanks @gluonhiggs for the quick fix!
I am not sure about this line where you are decomposing simple one-qubit gates such as "cx", "rx", "ry". Those gates are supported by Mitiq/Cirq and we risk to over-decompose the circuit.

I understand that tests are passing, but this change may unnecessarily increase the depth of the output circuit and this is a problem for the efficiency of the quantum computation. For example, if you input a short qiskit circuit made by a single "rzz" gate and many "cx", "rx", "ry" , Mitiq would decompose all gates producing an extremely long output circuit.

I don't know if it works, but I would suggest,it  **after** the existing line `circuit=circuit.decompose()`,
to add something like this:

`circuit = circuit.decompose(gates_to_decompose=["rzz", "rzx", "rxx"], reps=10)`

In this way, Mitiq will first apply a general first decomposition step as usual.
Afterwards, if there are still "rzz", "rzx", "rxx" gates left in the circuit, it will simplify only those specific gates.

What do you think?

Reply:
@cosenal I thought about that and tried to `transpile` not to the maximum set of the common gates, but to the set of the gates that can be used to construct the others, i.e., `["u1", "u2", "u3", "cx", "rx", "ry", "rz"]`, then `decompose()`. I also thought that this would be efficient because `transpile` come with `optimization_method`. However, the same error still happens. I even tried with a broader set, e.g., `["u3", "u2", "u1", "cx", "id", "x", "y", "z", "h", "s", "t", "rx", "ry", "rz"]`, but nothing has changed. Generally, I think it would be hard to determine the set because there will be more `n-qubit` gates, right?
@andreamari If we choose the set gate to decompose like `["rzz", "rzx", "rxx"]`, I'm wonder if we also have to consider other combinations of `x`, `y`, and `z` here like `ryy`, `ryz`, and so on?

Reply:
What I am proposing is something like
```python
circuit = qiskit.transpile(
    circuit,
    basis_gates=QASM2_GATE_SET,
    optimization_level=0,       
)
 mitiq_circuit = from_qasm(qasm2.dumps(circuit))
 ```
_without_ the `decompose()`, because that would bring the gate set to another set that may not be supported by QASM serializer/parser.

I tried with `QASM2_GATE_SET = ["u1", "u2", "u3", "cx", "rx", "ry", "rz"]` and it seems to work fine. I don't think that's the maximal set though. Having the maximal QASM-supported set there is the best we can do. And something smaller than the maximal will be sub-optimal when it comes to the size of the output circuit (as Andrea pointed out.)

Reply:
Yeah. It works. Let me have some tests. @andreamari what do you think? I think we can increase the level of optimization or expose the option to the interface.

Reply:
Two approaches are possible:

1- Decomposing all the gates that **we don't like** (specified by `gates_to_decompose`) until all the gates we don't like are gone (usually it's enough to add `reps>5`. )

2- Transpiling, as suggested by @cosenal, into a get set that **we like** (e.g. `QASM2_GATE_SET` or a subset of `QASM2_GATE_SET` that is also compatible with Qasm-to-Cirq conversion).

Solution 1 is closer to the existing workflow and can be good compromise. However, Solution 2 looks better at least in theory. 

@gluonhiggs :  I think you are confusing `gates_to_decompose`  with `basis_gates` . The `gates_to_decompose` contains bad gates that we want to decompose. `basis_gates` contains good gates that we want to have in the circuit.  So, e.g., `gates_to_decompose` should not contain `"cx", "rx", "ry"` since these are good gates that we want to keep in the circuit.

@gluonhiggs Yes, I think `ryy, ryz` and any other gate that is not supported should also be in `gates_to_decompose` if you go for Solution 1.

Reply:
@andreamari Why would we need repeated decompose() if we add all gates we don't like in `gates_to_decompose`?

Reply:
@gluonhiggs Can you elaborate on the tests that you think don't pass with the approach I proposed?
Regarding Update 2, I am not sure about the difference between decompose and transpile in the case where you don't specify any target hardware architecture on the transpile. It's something we can ask Qiskit folks on their community call.

Reply:
@andreamari I can distinguish `gates_to_decompose` and `basis_gates`. Perhaps, the way I expressed my thoughts was not clear at some point. The problem here is for some reasons, when I tried to decompose the "bad gates" (different lists of `rii` with this [reference](https://github.com/Qiskit/qiskit/tree/main/qiskit/circuit/library/standard_gates)), the error similar to #2578 (the original issue) still happens. 
@cosenal Here is the short test summary from `pytest`:
```bash
================================================================================= short test summary info =================================================================================
FAILED mitiq/interface/mitiq_qiskit/tests/test_conversions_qiskit.py::test_convert_with_qft - AssertionError: assert False
FAILED mitiq/zne/scaling/tests/test_folding.py::test_convert_qiskit_to_mitiq_circuit_conversion_error - Failed: DID NOT RAISE <class 'mitiq.interface.conversions.CircuitConversionError'>
FAILED mitiq/zne/scaling/tests/test_folding.py::test_folding_circuit_conversion_error_qiskit[fold_gates_at_random] - Failed: DID NOT RAISE <class 'mitiq.interface.conversions.CircuitConversionError'>
FAILED mitiq/zne/scaling/tests/test_folding.py::test_folding_circuit_conversion_error_qiskit[fold_global] - Failed: DID NOT RAISE <class 'mitiq.interface.conversions.CircuitConversionError'>
```
Should I paste the whole log here or move the discussion to discord?

Reply:
Here are the tests failed:
Solution 1:
mitiq/interface/mitiq_qiskit/tests/test_conversions_qiskit.py::test_convert_with_qft
mitiq/zne/tests/test_zne.py::test_execute_zne_on_qiskit_circuit_with_QFT

Solution 2:
mitiq/interface/mitiq_qiskit/tests/test_conversions_qiskit.py::test_convert_with_qft
mitiq/zne/scaling/tests/test_folding.py::test_convert_qiskit_to_mitiq_circuit_conversion_error
mitiq/zne/scaling/tests/test_folding.py::test_folding_circuit_conversion_error_qiskit[fold_gates_at_random]
mitiq/zne/scaling/tests/test_folding.py::test_folding_circuit_conversion_error_qiskit[fold_global]
The error log is too lengthy that I can't even paste to discord.

Reply:
> @andreamari Why would we need repeated decompose() if we add all gates we don't like in `gates_to_decompose`?

I think you are right, if all bad gates are in gates_to_decompose you probably don't need to also keep the existing line circuit.decompose() (without options).

However I think you would still need the `reps=...` option, since it ensures that potential bad gates that may be created  during the decomposition process are further decomposed.

Reply:
FYI: I have updated the code and it seems to work fine thanks to your suggestion @andreamari 
The story is:
I fell in the 2nd approach that used `transpile` to transform the circuits to some pre-declared `basis_gates`. I had tried dozens of combinations of the supported gates to form `basis_gates`. In the end, I realized that a very good choice of `basis_gates` passing almost all test cases but still can cause `transpile` to fail these tests
```bash
mitiq/zne/scaling/tests/test_folding.py::test_convert_qiskit_to_mitiq_circuit_conversion_error
mitiq/zne/scaling/tests/test_folding.py::test_folding_circuit_conversion_error_qiskit[fold_gates_at_random]
mitiq/zne/scaling/tests/test_folding.py::test_folding_circuit_conversion_error_qiskit[fold_global]
```
because `transpile` is too strong that can transform any circuit of both supported and unsupported gates to a circuit constructed from `basis_gates`. In the context of mitiq, to my understanding, we do not implement the noise mitigation for unsupported gates (or custom gates in the failed test) directly. Therefore, we expect that `convert_to_mitiq` will raise error in case there are custom gates, instead of transforming the circuit normally. So, this is too much. We should not use `transpile` here.

Back to the 1st approach, when these tests are failed
```bash
mitiq/interface/mitiq_qiskit/tests/test_conversions_qiskit.py::test_convert_with_qft
mitiq/zne/tests/test_zne.py::test_execute_zne_on_qiskit_circuit_with_QFT
```
so I tried to add `"QFT"` to the `gates_to_decompose` and it's worked since then. Previously, instead of using `"QFT"`, I used `"qft"`. That's why it didn't work.

Reply:
@andreamari 
>  since it ensures that potential bad gates that may be created during the decomposition process

I would be surprised if gates from the `gates_to_decompose` reappear during decomposition. Something we can ask to Qiskit folks.

Reply:
@gluonhiggs if we go with the approach I see now in the PR (good that it works), we could do the decomposition directly instead of doing it as a fallback when the conversion fails.